src/main/kotlin/domain/usecase/GetSeafoodMealsUseCase.kt

```diff
@@ -0,0 +1,21 @@
+package org.example.domain.usecase
+
+import org.example.domain.FoodException
+import org.example.domain.repository.FoodRepository
+
+class GetSeafoodMealsUseCase(
+    private val foodRepository: FoodRepository
+) {
+    fun execute(): List<Pair<String, Float>> {
```

Comment:
What about using the Result class ? it will be more readable in the UI
```suggestion
    fun execute(): Result<List<Pair<String, Float>>> {
```

Reply:
Result if u will return 2 different responses but in my case I return list as success and throw exception incase of there is no data.

What I see as the best to return the list in the 2 cases so it contains data or is empty.

but as we have now exception I prefer to throw exception to deal with exception now but for sure I prefer the Result more than exception